pkg/controller/common/reconciler.go

```diff
@@ -87,9 +88,111 @@ func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci network
 	return nil
 }
 
+func (r *BaseIngressReconciler) reconcileNetworkPolicy(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
+	desired := resources.MakeNetworkPolicyAllowAll(ci)
+
+	var networkPolicyList networkingv1.NetworkPolicyList
+	if err := r.Client.List(ctx, &client.ListOptions{}, &networkPolicyList); err != nil {
+		return err
+	}
+
+	// Detect if the user has any NetworkPolicy objects in this namespace
+	foundUserNetworkPolicy := false
+	for _, networkPolicy := range networkPolicyList.Items {
+		// Don't treat the NetworkPolicy we create as user-created
+		if networkPolicy.Name == desired.Name && networkPolicy.Namespace == desired.Namespace {
+			continue
+		}
+
+		// Don't treat the NetworkPolicy owned by Service Mesh as user-created
+		if networkPolicy.Labels["maistra.io/owner"] != "" {
+			continue
+		}
+
+		// Treat any other NetworkPolicy we find as user-created
+		foundUserNetworkPolicy = true
+	}
+
+	// If the user has created NetworkPolicy objects in this namespace
+	// then assume they are managing NetworkPolicy and do not create
+	// our own. If we previously created one and a user starts
+	// managing NetworkPolicy explicitly then this will allow them to
+	// delete ours without us automatically recreating it again.
+	if foundUserNetworkPolicy {
+		return nil
+	}
+
+	if err := r.ensureNetworkPolicy(ctx, ci, desired); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *BaseIngressReconciler) ensureNetworkPolicy(ctx context.Context, ci networkingv1alpha1.IngressAccessor, desired *networkingv1.NetworkPolicy) error {
+	logger := logging.FromContext(ctx)
+
+	networkPolicy := &networkingv1.NetworkPolicy{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, networkPolicy)
+
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
```suggestion
	if errors.IsNotFound(err) {
```

A bit simpler and semantically equivalent IIRC.